Assignment-clonecoding/Assignment-clonecoding/2nd/LoginView.swift

```diff
@@ -0,0 +1,323 @@
+        return button
+    }()
+    
+    private let separatorLabel: UILabel = {
+        let label = UILabel()
+        label.text = "|"
+        label.textColor = UIColor(red: 46/255, green: 46/255, blue: 46/255, alpha: 1)
+        label.font = UIFont(name: "Pretendard", size: 14)
+        label.textAlignment = .center
+        return label
+    }()
+
+    
+    private let findPW: UIButton = {
+        let button = UIButton()
+        button.setTitle("비밀번호 찾기", for: .normal)
+        button.titleLabel?.font = UIFont(name:"Pretendard-Light", size: 14)
+        button.setTitleColor(UIColor(red: 156/255, green: 156/255, blue: 156/255, alpha: 1), for: .normal)
+        return button
+    }()
+    
+    private let noAccount: UILabel = {
+        let label = UILabel()
+        label.text = "아직 계정이 없으신가요?"
+        label.font = UIFont(name:"Pretendard-Thin", size: 14)
+        label.textColor = UIColor(red: 98/255, green: 98/255, blue: 98/255, alpha: 1)
+        label.textAlignment = .center
+        return label
+    }()
+    
+    
+    private let createNickname: UIButton = {
+        let button = UIButton()
+        let attributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont(name: "Pretendard", size: 14) ?? UIFont.systemFont(ofSize: 14),
+            .foregroundColor: UIColor(red: 156/255, green: 156/255, blue: 156/255, alpha: 1),
+            .underlineStyle: NSUnderlineStyle.single.rawValue
+        ]
+        
+        let attributedTitle = NSAttributedString(string: "닉네임 만들러가기", attributes: attributes)
+        button.setAttributedTitle(attributedTitle, for: .normal)
+        
+        return button
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        idTextField.delegate = self
+        passwordTextField.delegate = self
+        
+        self.view.backgroundColor = .black
+        setLayout()
+        updateLoginButtonState()
+    }
+    
+    @objc func loginButtonDidTap() {
+        guard let id = idTextField.text, !id.isEmpty else {
+            return
+        }
+        
+        let welcomeViewController = WelcomeViewController()
+        welcomeViewController.id = id
+        if let navigationController = self.navigationController {
+            navigationController.pushViewController(welcomeViewController, animated: true)
+        } else {
+
+        }
+    }
+
+    
+    private func setLayout() {
+        [titleLabel, idTextField, passwordTextField, loginButton, findID, findPW, separatorLabel, noAccount, createNickname].forEach {
+            self.view.addSubview($0)
+        }
+        
+        titleLabel.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().offset(90)
+            make.left.equalToSuperview().inset(109)
+            make.right.equalToSuperview().inset(110)
+            make.width.equalTo(156)
```

Comment:
left와 right를 화면으로부터 얼만큼 여백이 남는지 정해주면 자동으로 넓이가 설저오디기 때문에 
```suggestion
            make.left.equalToSuperview().inset(109)
            make.right.equalToSuperview().inset(110)
```
width는 따로 설정해주시지 않아도 될 것 같습니다!